ui/app/components/app/modals/threebox-restore-confirm/threebox-restore-confirm.component.js

```diff
@@ -0,0 +1,58 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import Modal, { ModalContent } from '../../modal'
+
+export default class ThreeBoxRestoreConfirm extends PureComponent {
+  static propTypes = {
+    hideModal: PropTypes.func.isRequired,
+    restoreFromThreeBox: PropTypes.func.isRequired,
+    setThreeBoxSyncingPermission: PropTypes.func.isRequired,
+    setRestoredFromThreeBox: PropTypes.func.isRequired,
+    address: PropTypes.string.isRequired,
+  }
+
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  handleCancel = () => {
+    const { setRestoredFromThreeBox, hideModal } = this.props
+    setRestoredFromThreeBox(false)
+    hideModal()
+  }
+
+  handleConfirm = () => {
+    const {
+      restoreFromThreeBox,
+      setRestoredFromThreeBox,
+      setThreeBoxSyncingPermission,
+      hideModal,
+      address,
+    } = this.props
+    restoreFromThreeBox(address)
+      .then(() => {
+        setRestoredFromThreeBox(true)
+        setThreeBoxSyncingPermission(true)
```

Comment:
I'm not sure whether this is needed either. Isn't it impossible for `restoreFromThreeBox` to succeed unless 3box was already initialized, and doesn't it being initialized imply that 3box was already allowed?

Reply:
Hmm... that looks to be the case as well. What we do need here is to `_registerUpdates()`

Reply:
which is wrapped with the "public" method `turnThreeBoxSyncingOn()`

Reply:
Ah I see, makes sense.